block_priofees/src/block_priofees.rs

```diff
@@ -0,0 +1,123 @@
+use crate::rpc_data::{PrioFeesStats, PrioFeesUpdateMessage};
+use crate::stats_calculation::calculate_supp_stats;
+use log::{error, info, trace, warn};
+use solana_lite_rpc_core::types::BlockStream;
+use solana_sdk::clock::Slot;
+use std::collections::BTreeMap;
+use std::sync::Arc;
+use tokio::sync::broadcast::error::RecvError::{Closed, Lagged};
+use tokio::sync::broadcast::Sender;
+use tokio::sync::RwLock;
+use tokio::task::JoinHandle;
+
+// note: ATM only the latest slot (highest key) is used
+const SLOTS_TO_RETAIN: u64 = 100;
+
+/// put everything required to serve sync data calls here
+#[derive(Clone)]
+pub struct PrioFeeStore {
+    // store priofees stats for recently processed blocks up to CLEANUP_SLOTS_AFTER
+    recent: Arc<RwLock<BTreeMap<Slot, PrioFeesStats>>>,
+}
+
+pub struct PrioFeesService {
+    pub block_fees_store: PrioFeeStore,
+    // use .subscribe() to get a receiver
+    pub block_fees_stream: Sender<PrioFeesUpdateMessage>,
+}
+
+impl PrioFeesService {
+    pub async fn get_latest_priofees(&self) -> Option<(Slot, PrioFeesStats)> {
+        let lock = self.block_fees_store.recent.read().await;
+        let latest_in_store = lock.last_key_value();
+        latest_in_store.map(|x| (*x.0, x.1.clone()))
+    }
+}
+
+pub async fn start_block_priofees_task(
+    mut block_stream: BlockStream,
+) -> (JoinHandle<()>, PrioFeesService) {
+    let recent_data = Arc::new(RwLock::new(BTreeMap::new()));
+    let store = PrioFeeStore {
+        recent: recent_data.clone(),
+    };
+    let (priofees_update_sender, _priofees_update_receiver) = tokio::sync::broadcast::channel(64);
+    let sender_to_return = priofees_update_sender.clone();
+
+    let jh_priofees_task = tokio::spawn(async move {
+        let sender = priofees_update_sender.clone();
+        'recv_loop: loop {
+            let block = block_stream.recv().await;
+            match block {
+                Ok(block) => {
+                    {
+                        // first do some cleanup
+                        let mut lock = recent_data.write().await;
+                        lock.retain(|slot, _| *slot > slot - SLOTS_TO_RETAIN);
```

Comment:
Do saturating sub, could over flow for local cluster.

Reply:
good catch!